File: src/main/java/org/akkreditierung/ui/model/AdvertiserConfig.java
package org.akkreditierung.ui.model;

import java.io.Serializable;
import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;

/**
 * @author frank
 * 
 */
@Entity
@Table(name = "studiengaenge")
public class AdvertiserConfig implements Serializable {

	private static final long serialVersionUID = -1587664618577852245L;
	@Id
	private int id;
	private String fach;
    private String abschluss;
    private String hochschule;
    private String bezugstyp;

    public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getFach() {
		return fach;
	}

	public void setFach(String fach) {
		this.fach = fach;
	}

    public String getAbschluss() {
        return abschluss;
    }

    public void setAbschluss(String abschluss) {
        this.abschluss = abschluss;
    }

    public String getHochschule() {
        return hochschule;
    }

    public void setHochschule(String hochschule) {
        this.hochschule = hochschule;
    }

    public String getBezugstyp() {
        return bezugstyp;
    }

    public void setBezugstyp(String bezugstyp) {
        this.bezugstyp = bezugstyp;
    }
}

File: src/main/scala/org/akkreditierung/ui/model/AdvertiserConfig.scala
package org.akkreditierung.ui.model

import javax.persistence._
import java.util.{Date, Map}
import scala.beans.BeanProperty
import org.akkreditierung.model.Studiengang

@Entity
@Table(name = "studiengaenge") class AdvertiserConfig {
  @Id @BeanProperty var id: Int = 0
  @BeanProperty var fach: String = null
  @BeanProperty @Column(name = "jobId") var jobId: Int = 0
  @BeanProperty var abschluss: String = null
  @BeanProperty var hochschule: String = null
  @BeanProperty var bezugstyp: String = null
  @BeanProperty var link: String = null
  @BeanProperty @Column(name = "GutachtenLink") var gutachtenLink: String = null
  @BeanProperty @Column(name = "modifiedDate") var modifiedDate: Date = null
  @BeanProperty @Column(name = "sourceId") var sourceId: Int = 0
  @OneToMany(cascade = Array(CascadeType.ALL), fetch = FetchType.LAZY)
  @JoinColumn(name = "id", referencedColumnName = "id", nullable = true)
  @MapKey(name = "k")
  @BeanProperty var map: Map[String, StudiengaengeAttribute] = null

//  override def toString: String = {
//    "studiengaenge{" + "id=" + id + ", fach='" + fach + '}'
//  }

  def toStudienGang() = {
    Studiengang(jobId=Option(jobId), fach=fach, abschluss=abschluss, hochschule=hochschule, bezugstyp=bezugstyp, link=Option(link), gutachtentLink=Option(gutachtenLink), modifiedDate = Option(modifiedDate), sourceId = sourceId)
  }
}